rs/nns/governance/src/heap_governance_data.rs

```diff
@@ -84,16 +84,35 @@ impl From<XdrConversionRate> for XdrConversionRatePb {
     }
 }
 
+/// Converts a vector of u8s to array of length 32, which is the length needed for our rng seed.
+/// If the array is the wrong size, this returns an error.
+fn vec_to_array(v: Vec<u8>) -> Result<[u8; 32], String> {
+    let boxed_slice = v.into_boxed_slice();
+    let boxed_array: Box<[u8; 32]> = match boxed_slice.try_into() {
+        Ok(hash) => hash,
+        Err(original) => {
+            return Err(format!(
+                "Expected a hash of length {} but it was {}",
+                32,
+                original.len()
+            ))
+        }
+    };
+    Ok(*boxed_array)
+}
```

Comment:
I think there is a `TryFrom` implementation that has nearly this signature - might be simpler to use that:

```suggestion
fn vec_to_array(v: Vec<u8>) -> Result<[u8; 32], String> {
   [u8; 32]::try_from(v).map_err(|_| format!(
                "Expected a hash of length {} but it was {}",
                32,
                original.len()
            ))
}
```